pkg/apis/autoscaling/validation/validation.go

```diff
@@ -342,6 +359,32 @@ func validatePodsSource(src *autoscaling.PodsMetricSource, fldPath *field.Path)
 	return allErrs
 }
 
+func validateContainerResourceSource(src *autoscaling.ContainerResourceMetricSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(src.Name) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("name"), "must specify a resource name"))
```

Comment:
think about what bounds we place on all new fields. e.g. should this be length-constrained? limited to specific characters?

Reply:
I have changed it to use an existing helper method which validates that the resource is a valid container resource.